bitfield/impl/src/bitfield.rs

```diff
@@ -0,0 +1,245 @@
+use proc_macro::TokenStream;
+use proc_macro2::TokenStream as TokenStream2;
+use syn::{
+    self,
+    Token,
+    punctuated::Punctuated,
+    spanned::Spanned as _,
+    parse::{
+        Parse,
+        ParseStream,
+        Result as ParseResult,
+    }
+};
+use quote::{
+    quote,
+    quote_spanned,
+    ToTokens,
+};
+
+pub fn generate(args: TokenStream, input: TokenStream) -> TokenStream {
+    match bitfield_impl(args.into(), input.into()) {
+        Ok(output) => output.into(),
+        Err(err) => err.to_compile_error().into(),
+    }
+}
+
+fn bitfield_impl(args: TokenStream2, input: TokenStream2) -> Result<TokenStream2, syn::Error> {
+    let _ = args;
+    let input = syn::parse::<BitfieldStruct>(input.into())?;
```

Comment:
You might be right. The only reason I do it via `BitfieldStruct` is so that I can have methods instead of free functions. But in the end it comes down to taste.